hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -1050,7 +1050,7 @@ public void run() {
         }
       }
 
-      // We registered with the Master.  Go into run mode.
+      // Run mode.
       long lastMsg = System.currentTimeMillis();
```

Comment:
How about setting this lastMsg here should be set to Long.MAX_LONG so we heartbeat immediately after the report-for-duty... so no lag before the RS is 'online'?

Reply:
I think you mean set it to `LONG.MIN_LONG` (or 0), since the line that checks it says `if ((now - lastMsg) >= msgInterval)`, but yes good one 👍

Reply:
You are right.

Reply:
So, do in a follow-on @caroliney14  ?

Reply:
@saintstack do we have to do this one? asking because the `lastMsg` timestamp ultimately ends up getting stored [here](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java#L1425), not sure if it'll have any impact we care about. also, one stage of the PR build kept failing for some unknown reason when I had pushed up the change `long lastMsg = 0;` which is why I ended up undoing it

Reply:
No. We not have to do it. It was suggestion.  You've done a mountain of great work in here already.

I'll keep an eye on it. Was thinking we heartbeat immediately but that might not be a good idea on a big cluster....

Reply:
sounds good, thank you @saintstack